### PR TITLE
Support specifying position model/frame of reference for lighting instances

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR introduces JSON support, and appropriate testing and documentation, for setting the desired position model/frame of reference for a lighting instance.  With this setting, the user can specify whether they wish for a particular light info/instance to be placed in the scene relative to the global origin or the camera.  Object-relative positioning is also fully supported in the configuration, but only as a stub currently in the underlying lighting code.

The enum describing the position model value has also been refactored to follow our naming conventions, from  ALL_CAPS to pascal case, in the c++ and python code.
 

## How Has This Been Tested
All existing and added c++ and python test cases pass

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by aclegg3 — APPROVED** [review]

File: src/esp/gfx/LightSetup.cpp [hunk 1]
```diff
@@ -28,11 +28,11 @@ Magnum::Vector4 getLightPositionRelativeToCamera(
                         "w == 1 for a point light");
 
   switch (light.model) {
-    case LightPositionModel::OBJECT:
+    case LightPositionModel::Object:
```
Comment (Skylion007) [inline src/esp/gfx/LightSetup.cpp:31]: Why the change in style?

Comment (jturner65) [reply src/esp/gfx/LightSetup.cpp:31]: This was a refactor in other enums through Habitat-Sim we decided [upon a few months ago](https://github.com/facebookresearch/habitat-sim/pull/1009).  Guess this enum got missed.

File: docs/pages/attributesJSON.rst [hunk 2]
```diff
@@ -230,3 +230,6 @@ The Light Setup attributes JSON should contain a single cell named "lights" that
 "type"
 	- string
 	- The type of the light.  "point" and "directional" are currently supported.
+"position_model"
+  - string
+  - They frame to use to place the light. "global", meaning stage's origin, and "camera", meaning place relative to a (potentially moving) camera, are currently supported.
```
Comment (bigbike) [inline docs/pages/attributesJSON.rst:235]: Oh, the "object" mode cannot be supported at the moment.

**Review by bigbike — COMMENTED** [review]
This is great! Thank you very much!

File: docs/pages/attributesJSON.rst (hunk 2 above)
Comment (jturner65) [reply docs/pages/attributesJSON.rst:235]: yes, this would require more configuration specifications at the scene instance level, to specify which object instance the light should "orbit"

Comment (bigbike) [reply docs/pages/attributesJSON.rst:235]: Not "orbit". Just stay fixed relative to the object.
In that case, the "light" should be an object in the scene graph, and be attached to a child node of that object.

Comment (jturner65) [reply docs/pages/attributesJSON.rst:235]: Fixed orbit :).
